realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -557,6 +616,10 @@ private void setValue(String fieldName, Object value) {
         } else if (valueClass == RealmList.class) {
             RealmList<?> list = (RealmList<?>) value;
             setList(fieldName, list);
+        } else if (valueClass == Decimal128.class) {
```

Comment:
As discussed, let's add the bson dependency for this PR and optimise later a mechanism to make it optional. I think the method count and size will not affect apps not using these types (because of R8 tree shaking)